### ISSUE TITLE
Error to build on Windows: Cannot find module 'rimraf'

### ISSUE THREAD

**Issue by adelarsq** [onivim/oni#918]
When compiling on Windows occurs this error:

```
C:\dev\oni {git}{hg}
{lamb} npm install   && npm run build   && npm link   && oni

> oni-neovim-binaries@0.0.6 postinstall C:\dev\oni\node_modules\oni-neovim-binaries
> node lib/postinstall.js

module.js:487
    throw err;
    ^

Error: Cannot find module 'rimraf'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\dev\oni\node_modules\oni-neovim-binaries\lib\postinstall.js:12:16)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: 7zip-bin-linux@1.1.0 (node_modules\7zip-bin-linux):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for 7zip-bin-linux@1.1.0: wanted {"os":"linux","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: 7zip-bin-mac@1.0.1 (node_modules\7zip-bin-mac):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for 7zip-bin-mac@1.0.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! oni-neovim-binaries@0.0.6 postinstall: `node lib/postinstall.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the oni-neovim-binaries@0.0.6 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\myuser\AppData\Roaming\npm-cache\_logs\2017-11-10T13_51_42_964Z-debug.log
```

**Comment by bryphe** [comment]
Weird! Thanks for the logs. Just curious - what version of `npm` and `node` are you using?

You can check via:
```
npm --version
node --version
```

I had to upgrade the travis CI and appveyor machines to Node 8 to get all the dependencies + successfully npm install, perhaps that is the issue?

**Comment by adelarsq** [comment]
`npm` and `node` versions:
```
npm --version
5.5.1
node --version
v8.1.3
```
More logs:
```
59032 verbose lifecycle oni-neovim-binaries@0.0.7~postinstall: CWD: C:\dev\oni\node_modules\oni-neovim-binaries
59033 silly lifecycle oni-neovim-binaries@0.0.7~postinstall: Args: [ '/d /s /c', 'node lib/postinstall.js' ]
59034 silly lifecycle oni-neovim-binaries@0.0.7~postinstall: Returned: code: 1  signal: null
59035 info lifecycle oni-neovim-binaries@0.0.7~postinstall: Failed to exec postinstall script
59036 verbose unlock done using C:\Users\my.user\AppData\Roaming\npm-cache\_locks\staging-b80d5a986906776c.lock for C:\dev\oni\node_modules\.staging
59037 verbose stack Error: oni-neovim-binaries@0.0.7 postinstall: `node lib/postinstall.js`
59037 verbose stack Exit status 1
59037 verbose stack     at EventEmitter.<anonymous> (C:\Users\my.user\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:280:16)
59037 verbose stack     at emitTwo (events.js:125:13)
59037 verbose stack     at EventEmitter.emit (events.js:213:7)
59037 verbose stack     at ChildProcess.<anonymous> (C:\Users\my.user\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
59037 verbose stack     at emitTwo (events.js:125:13)
59037 verbose stack     at ChildProcess.emit (events.js:213:7)
59037 verbose stack     at maybeClose (internal/child_process.js:897:16)
59037 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:208:5)
59038 verbose pkgid oni-neovim-binaries@0.0.7
59039 verbose cwd C:\dev\oni
59040 verbose Windows_NT 10.0.10586
59041 verbose argv "C:\\dev\\nodejs\\node.exe" "C:\\Users\\my.user\\AppData\\Roaming\\npm\\node_modules\\npm\\bin\\npm-cli.js" "install"
59042 verbose node v8.1.3
59043 verbose npm  v5.5.1
59044 error code ELIFECYCLE
59045 error errno 1
59046 error oni-neovim-binaries@0.0.7 postinstall: `node lib/postinstall.js`
59046 error Exit status 1
59047 error Failed at the oni-neovim-binaries@0.0.7 postinstall script.
59047 error This is probably not a problem with npm. There is likely additional logging output above.
59048 verbose exit [ 1, true ]
```
Its something related with https://github.com/dbashford/mimosa/issues/374 but cleaning the cache and removing the `node_modules` folder didn't solve.

**Comment by adelarsq** [comment]
@bryphe to sucessfully build on Windows I had to execute:
```
npm clean cache --force
npm install --force rimraf
npm install --force fs-extra
```
Related issues: https://github.com/npm/npm/issues/17747 https://github.com/angular/angular/issues/5343

**Comment by bryphe** [comment]
Ah ok, thanks @adelarsq ! I'll add these as top-level dependencies in #931 in order to improve the build reliability.

**Comment by bryphe** [comment]
Should be addressed now on master by #931 , as I've included rimraf and fs-extra explicitly in the package.json for Oni. Thanks for the investigation, @adelarsq !
